go/arrow/internal/testing/types/extension_types.go

```diff
@@ -18,20 +18,171 @@
 package types
 
 import (
+	"bytes"
 	"encoding/binary"
 	"fmt"
 	"reflect"
+	"strings"
+
+	"github.com/goccy/go-json"
 
 	"github.com/apache/arrow/go/v12/arrow"
 	"github.com/apache/arrow/go/v12/arrow/array"
+	"github.com/google/uuid"
 	"golang.org/x/xerrors"
 )
 
+type UUIDBuilder struct {
+	*array.ExtensionBuilder
+}
+
+func NewUUIDBuilder(bldr *array.ExtensionBuilder) *UUIDBuilder {
+	b := &UUIDBuilder{
+		ExtensionBuilder: bldr,
+	}
+	return b
+}
+
+func (b *UUIDBuilder) Append(v uuid.UUID) {
+	b.ExtensionBuilder.Builder.(*array.FixedSizeBinaryBuilder).Append(v[:])
+}
+
+func (b *UUIDBuilder) UnsafeAppend(v uuid.UUID) {
+	b.ExtensionBuilder.Builder.(*array.FixedSizeBinaryBuilder).UnsafeAppend(v[:])
+}
+
+func (b *UUIDBuilder) AppendValues(v []uuid.UUID, valid []bool) {
+	data := make([][]byte, len(v))
+	for i, v := range v {
+		data[i] = v[:]
+	}
+	b.ExtensionBuilder.Builder.(*array.FixedSizeBinaryBuilder).AppendValues(data, valid)
+}
+
+func (b *UUIDBuilder) UnmarshalOne(dec *json.Decoder) error {
+	t, err := dec.Token()
+	if err != nil {
+		return err
+	}
+
+	var val uuid.UUID
+	switch v := t.(type) {
+	case string:
+		data, err := uuid.Parse(v)
+		if err != nil {
+			return err
+		}
+		val = data
+	case []byte:
+		data, err := uuid.ParseBytes(v)
+		if err != nil {
+			return err
+		}
+		val = data
```

Comment:
Is this case actually getting tested by the tests you wrote? Check the test coverage please to confirm. I think it would only go through the String case. I'm not positive though.

Reply:
I just copied that from other places in the code but seems if I remove it then all is good.